### PR TITLE
KNE-25: JSAPI client auth middleware refactoring

### PR DESCRIPTION
**Pull request by jgpacheco**

### CODE REVIEW

File: submodules/kneJsapi.js [hunk 1]
```diff
@@ -243,11 +246,15 @@ KneJsapi.prototype.getClient = function () {
...
+    var allowTokenRefresh = true;
+
+    try { allowTokenRefresh = !!self.engine.getConfig('kne-jsapi.allowTokenRefreshOnAuthenticationMiddleware'); }
+    catch (e) { allowTokenRefresh = true; }
 
     if (socket.handshake.query.token) {
         self.wrapper(rest.get.bind(self)(self.engine.getConfig('kne-jsapi.url') + '/me', {
             accessToken: socket.handshake.query.token
-        })).then(function (res) {
+        }), { 'allowTokenRefresh': allowTokenRefresh }).then(function (res) {
```
Comment (rossnelson) [inline submodules/kneJsapi.js]: A socket connection token should never be refreshed with a server side token.

Comment (jgpacheco) [inline submodules/kneJsapi.js]: You are suggesting set default value to `false` or the removal of the configuration option? I think this way is flexible enough to manage future scenarios.

Comment (rossnelson) [inline submodules/kneJsapi.js]: I can see that. I guess I'm fine with this implementation. I just can't see why `wrapper` would be needed in the middleware function. It only updates the client token and the middleware only ever accesses the `me` endpoint using a user token.

Comment (jgpacheco) [inline submodules/kneJsapi.js]: I will check again but  think that the `wrapper` function is used for all JSAPI calls.

**Pull request closed without merging**